### PR TITLE
Add built in labels to CloudRun revision

### PR DESCRIPTION
**Pull request by knanao**
**What this PR does / why we need it**:
Add built-in-labels to CloudRun revision in order to filter revisions by revision name.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/app/piped/executor/cloudrun/cloudrun.go [hunk 1]
```diff
@@ -148,12 +148,21 @@ func revisionExists(ctx context.Context, client provider.Client, revisionName st
 	return false, err
 }
 
-func addBuiltinLabels(sm provider.ServiceManifest, hash, pipedID, appID string) {
+func addBuiltinLabels(sm provider.ServiceManifest, lp executor.LogPersister, hash, pipedID, appID, revisionName string) bool {
```
Comment (khanhtc1202) [inline pkg/app/piped/executor/cloudrun/cloudrun.go:151]: should keep the `lp executor.LogPersister` as the last parameter, just as other functions (can be counted as our convention)
```suggestion
func addBuiltinLabels(sm provider.ServiceManifest, hash, pipedID, appID, revisionName string, lp executor.LogPersister) bool {
```

Comment (knanao) [reply pkg/app/piped/executor/cloudrun/cloudrun.go:151]: I see! let me fix it:D

File: pkg/app/piped/executor/cloudrun/cloudrun.go [hunk 2]
```diff
@@ -148,12 +148,21 @@ func revisionExists(ctx context.Context, client provider.Client, revisionName st
...
 		provider.LabelPiped:       pipedID,
 		provider.LabelApplication: appID,
 		provider.LabelCommitHash:  hash,
 	}
+	// Set builtinLabels for Service.
 	sm.AddLabels(labels)
+
+	// Set buildinLabels for Revision.
+	labels[provider.LabelRevisionName] = revisionName
+	if err := sm.AddRevisionLabels(labels); err != nil {
```
Comment (khanhtc1202) [inline pkg/app/piped/executor/cloudrun/cloudrun.go:163]: I'm a bit worried here, looks like the labels pointer changed after this L162 (add one more key) and we do that on purpose in order to make sure `RevisionLabels` and `Labels` are different, right?
But with the same labels object, change on L162 will affect the labels we added to sm.Labels (ref: https://go.dev/play/p/dy5xoUlFINK). Sorry if I missed something, but I guess better to add test for this function to ensure we get what we want, it's a standalone function, so looks like adding test for it is not too annoying any way 😄

Comment (knanao) [reply pkg/app/piped/executor/cloudrun/cloudrun.go:163]: @khanhtc1202 
Thank you for your reviewing!
Sorry, I couldn't got you perfectly, but I added test.
With that, wdyt?

**Comment by khanhtc1202** [comment]
Nice, thank you 👍 
/lgtm

**Comment by nghialv** [comment]
Nice.
/approve

**Pull request merged**